From c4d4e99945d9c0eeb4830f6b5bff789a03bc2ad0 Mon Sep 17 00:00:00 2001
From: Arseniy Velikanov <adomerlee@gmail.com>
Date: Mon, 14 Aug 2023 18:38:03 +0400
Subject: [PATCH 15/91] fixup! arm64: dts: mediatek: mt6765: Add USB

Use clocks from downstream kernel
Fixed usb-phy regs (taken from datasheet)

Signed-off-by: Arseniy Velikanov <adomerlee@gmail.com>
---
 arch/arm64/boot/dts/mediatek/mt6765.dtsi | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt6765.dtsi b/arch/arm64/boot/dts/mediatek/mt6765.dtsi
index a4b577029..941b2562d 100644
--- a/arch/arm64/boot/dts/mediatek/mt6765.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt6765.dtsi
@@ -201,25 +201,25 @@ usb2: usb@11200000 {
             reg = <0 0x11200000 0 0x1000>;
             interrupts = <GIC_SPI 73 IRQ_TYPE_LEVEL_LOW>;
             interrupt-names = "mc";
-            clocks = <&topckgen CLK_TOP_AXI_SEL>,
-                     <&topckgen CLK_TOP_AXI_SEL>,
+            clocks = <&infracfg CLK_IFR_ICUSB>,
+                     <&topckgen CLK_TOP_USB_TOP_SEL>,
                      <&topckgen CLK_TOP_UNIVPLL3_D4>;
             clock-names = "main", "mcu", "univpll";
             phys = <&u2port PHY_TYPE_USB2>;
             dr_mode = "peripheral";
         };
 
-        u2phy: t-phy@11210000 {
+        u2phy: t-phy@11CC0000 {
             compatible = "mediatek,mt6765-tphy",
                          "mediatek,generic-tphy-v1";
-            reg = <0 0x11210000 0 0x800>;
+            reg = <0 0x11CC0000 0 0x800>;
             #address-cells = <2>;
             #size-cells = <2>;
             ranges;
 
-            u2port: usb-phy@11210800 {
-                reg = <0 0x11210800 0 0x100>;
-                clocks = <&topckgen CLK_TOP_USB20_192M_D4>;
+            u2port: usb-phy@11CC0800 {
+                reg = <0 0x11CC0800 0 0x100>;
+                clocks = <&infracfg CLK_IFR_ICUSB>;
                 clock-names = "ref";
                 #phy-cells = <1>;
             };
